msal/src/main/java/com/microsoft/identity/client/internal/controllers/LocalMSALController.java

```diff
@@ -457,4 +461,35 @@ public boolean getDeviceMode(CommandParameters parameters) throws Exception {
     public boolean removeCurrentAccount(RemoveAccountCommandParameters parameters) throws Exception {
         return removeAccount(parameters);
     }
+
+    @Override
+    public AuthorizationResult deviceCodeFlowAuthRequest(final DeviceCodeFlowCommandParameters parameters) throws Exception {
+        // TODO: Placeholder to avoid inheritance error. Will be implemented after Command/Controller level PR in Common
+        return null;
+    }
+
+    @Override
+    public AcquireTokenResult acquireDeviceCodeFlowToken(final AuthorizationResult authorizationResult, DeviceCodeFlowCommandParameters commandParameters) throws Exception {
+        // TODO: Placeholder to avoid inheritance error. Will be implemented after Command/Controller level PR in Common
+        return null;
+    }
+
+    /**
+     * Helper method to check if a result object is valid (was a success). If not, an exception will be generated and thrown.
+     * @param result result object to be checked
+     * @throws MsalServiceException MsalServiceException object reflecting error code returned by the result
+     */
+    private void validateServiceResult(@NonNull IResult result) throws MsalServiceException {
+        //TODO: Will be implemented after Command/Controller level PR in Common
+    }
+
+    /**
+     * Given an error response object, create a serviceException object using the predefined error codes.
+     * @param response error response object to be checked
+     * @return an exception object
+     */
+    private ServiceException createServiceExceptionFromErrorResponse(IErrorResponse response) {
```

Comment:
should this return `MsalServiceException` instead? Also may be this should go in the `ExceptionAdapter` class?

Reply:
The conversion to `MsalServiceException` happens inside `LocalMsalController`. I made it return `ServiceException` so that it can be used in Broker later. I made it a private, local function since It's only being used for DCF currently, so placing in `ExceptionAdapter` may be a bit misleading.

Reply:
I will remove it from this PR and include it in the next one.